Assets/Project/Modules/Enemies/EnemyVFX/Scripts/EnemyVisuals.cs

```diff
@@ -36,23 +36,39 @@ private void Awake()
             }
         }
 
-        public void Configure()
+        public void Configure(IParticleFactory particleFactory)
         {
+            _particleFactory = particleFactory;
             _originalMeshDatas[0]._mesh.material.SetFloat("_Health", 1.0f);
         }
 
-        public virtual void PlayHitEffects(float healthCoef01)
+        public virtual void PlayHitEffects(float healthCoef01, DamageHit damageHit)
         {
             _originalMeshDatas[0]._mesh.material.SetFloat("_Health", healthCoef01);
-            
-            //smallest camera shake
 
+            ParticlesHitEffect(damageHit);
+            FlashHitEffect().Forget();
+        }
 
-            CircleEffect().Forget();
-            FlashEffect().Forget();
+        public virtual void PlayDeathEffects(DamageHit damageHit)
+        {
+            ParticlesHitEffect(damageHit);
+        }
+
+        private void ParticlesHitEffect(DamageHit damageHit)
+        {
+            //TODO: FIX these two things
+            //get the    position   of the contact point
+            //get the     normal    of the contact poiint
+            
+            Transform player = ServiceLocator.Instance.GetService<IGameReferences>().GetPlayer();
+            Vector3 spawnPos = transform.position + (player.position - transform.position) * 0.3f; //0.3f serves as enemy width
```

Comment:
(player.position - transform.position) isn't normalized, is it intended?

Reply:
It is not 🥶